src/ameedf/jb8223/assignments/hw02/Clock.java

```diff
@@ -34,4 +34,106 @@
  * BONUS 2: the same functionality as in "BONUS 1" while using as few member variables / memory as possible
  */
 public class Clock {
+
+    private int hours;
+    private int minutes;
+    private int seconds;
+    private int milliseconds;
+
+    private final int minValueAllowed = 0;
+    private final int maxValueAllowed = 1_000_000_000;
+
+    // 1st constructor, initializes the clock to 14:07:53.980
+    public Clock() {
+        this.hours = 14;
+        this.minutes = 7;
+        this.seconds = 53;
+        this.milliseconds = 980;
+    }
+
+    // 2nd constructor, accepts 4 parameters in the ranges of 0..23, 0..59, 0..59, 0..999
+    public Clock (int hours, int minutes, int seconds, int microSeconds) {
+        this.hours = hours;
+        this.minutes = minutes;
+        this.seconds = seconds;
+        this.milliseconds = microSeconds;
+    }
+
+    // addHours - with a single parameter which represents the number of hours to add.
+    public void addHours(int hoursToAdd) {
+        if (hoursToAdd >= minValueAllowed && hoursToAdd <= maxValueAllowed) {
+                this.hours = (this.hours + hoursToAdd) % 24;
+        }
+    }
+
+    // addMinutes - with a single parameter which represents the number of minutes to add.
+    public void addMinutes(int minutesToAdd) {
+        if (minutesToAdd >= minValueAllowed && minutesToAdd <= maxValueAllowed) {
+                addHours(minutesToAdd / 60);
```

Comment:
BUG: what happens if the clock is 10:59:12.013 and you want to add 2 minutes ? does the hours variable change?